File: 1. Two Sum.js
var twoSum = function (nums, target) {
  const numObj = {};

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];
    numObj[nums[i]] = i;
    if (numObj.hasOwnProperty(complement)) {
      return [numObj[complement], i];
    }
  }
  return [];
};

nums = [2, 7, 11, 15];
console.log(twoSum(nums, 26));

File: python/1.Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num_to_index = {}  # Dictionary to store value: index
        
        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_to_index:
                # Found the pair
                return [num_to_index[complement], i]
            # Store current number with its index
            num_to_index[num] = i
        
        # According to problem statement, we always have a solution
        # So this line won't be reached, but kept for completeness
        return []